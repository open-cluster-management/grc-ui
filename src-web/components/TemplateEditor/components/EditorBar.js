/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { Search } from 'carbon-components-react'
import '../scss/editor-bar.scss'
import '../../../../graphics/diagramIcons.svg'
import msgs from '../../../../nls/platform.properties'

class EditorBar extends React.Component {

  static propTypes = {
    handleEditorCommand: PropTypes.func,
    handleSearchChange: PropTypes.func,
    hasRedo: PropTypes.bool,
    hasUndo: PropTypes.bool,
    type: PropTypes.string,
  }

  constructor (props) {
    super(props)
    this.state = {
      searchName: '',
    }
  }

  handleClick = (command) => {
    this.props.handleEditorCommand(command)
  }

  handleSearch = ({target}) => {
    const searchName = target.value||''
    this.props.handleSearchChange(searchName)
    this.setState({searchName})
  }

  render() {
    const { locale } = this.context
    const { hasUndo, hasRedo, type } = this.props
    const { searchName } = this.state

    const undoButtons = [
      {command: 'undo', tooltip: msgs.get('editor.bar.undo', locale), icon:'undo', disabled:!hasUndo},
      {command: 'redo', tooltip: msgs.get('editor.bar.redo', locale), icon:'redo', disabled:!hasRedo},
    ]

    const nextButtons = [
      {command: 'previous', tooltip: msgs.get('editor.bar.previous', locale), icon:'previous', disabled:!searchName},
      {command: 'next', tooltip: msgs.get('editor.bar.next', locale), icon:'next', disabled:!searchName},
    ]

    const resetButtons = [
      {command: 'restore', tooltip: msgs.get('editor.bar.reset', locale), disabled:!hasUndo&&!hasRedo},
    ]

    const closeButtons = [
      {command: 'close', tooltip: msgs.get('editor.bar.close', locale), icon:'close'},
    ]

    const searchTitle = msgs.get('find.label', locale)
    return (
      <div className='editor-bar'>
        <div className='editor-bar-title'>
          {msgs.get('creation.view.yaml', locale)}
        </div>
        <div className='editor-bar-toolbar'>
          <div className='editor-bar-section'>
            {resetButtons.map((button) => {
              const {command} = button
              return (
                <EditorButton
                  key={command}
                  command={command}
                  button={button}
                  handleClick={this.handleClick}
                />
              )
            })}
          </div>
          <div className='editor-bar-section'>
            {undoButtons.map((button) => {
              const {command} = button
              return (
                <EditorButton
                  key={command}
                  command={command}
                  button={button}
                  handleClick={this.handleClick}
                />
              )
            })}
          </div>
          <div className='editor-bar-section'>
            <div className='editor-bar-search' role='region' aria-label={searchTitle} id={searchTitle}>
              <Search
                id={`template-editor-search-${type}`}
                labelText=''
                aria-label={searchTitle}
                placeHolderText={searchTitle}
                small={true} onChange={this.handleSearch}
              />
              {nextButtons.map((button) => {
                const {command} = button
                return (
                  <EditorButton
                    key={command}
                    command={command}
                    button={button}
                    handleClick={this.handleClick}
                  />
                )
              })}
            </div>
          </div>
          <div className='editor-bar-section'>
            {closeButtons.map((button) => {
              const {command} = button
              return (
                <EditorButton
                  key={command}
                  command={command}
                  button={button}
                  handleClick={this.handleClick}
                />
              )
            })}
          </div>
        </div>
      </div>
    )
  }
}

class EditorButton extends React.Component {
  static propTypes = {
    button: PropTypes.object,
    command: PropTypes.string,
    handleClick: PropTypes.func,
  }

  handleClick = () => {
    const {command, button: {disabled}} = this.props
    if (!disabled) {
      document.activeElement.blur()
      this.props.handleClick(command)
    }
  }

  handleKeyPress = (e) => {
    if ( e.key === 'Enter') {
      this.props.handleClick(this.props.command)
    }
  }

  render() {
    const {button: {disabled, tooltip, icon, className='', spacer}} = this.props
    if (spacer) {
      return (
        <div className='editor-bar-spacer'></div>
      )
    } else {
      const classes = classNames({
        'editor-bar-button': true,
        [`${className}`]: true,
        disabled
      })
      return (
        <div className={classes} tabIndex={0} role={'button'} aria-label={tooltip} title={tooltip}
          onClick={this.handleClick} onKeyPress={this.handleKeyPress} >
          {icon?
            <svg>
              <use href={`#diagramIcons_${icon}`} ></use>
            </svg> :
            <div>{tooltip}</div>
          }
        </div>
      )
    }
  }
}

export default EditorBar
