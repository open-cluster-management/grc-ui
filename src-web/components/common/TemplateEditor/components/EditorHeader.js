/*******************************************************************************
 * Licensed Materials - Property of IBM
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import '../scss/editor-header.scss'
import msgs from '../../../../nls/platform.properties'

class EditorHeader extends React.Component {

  static propTypes = {
    children: PropTypes.node,
    locale: PropTypes.string,
  }

  constructor (props) {
    super(props)
  }

  render() {
    const { children, locale } = this.props
    const editorToolbarTitle = msgs.get('editor.toolbar', locale)
    return (
      <div className='creation-view-yaml-header' >
        <div className='creation-view-yaml-header-toolbar' role='region' aria-label={editorToolbarTitle} id={editorToolbarTitle}>
          {children}
        </div>
      </div>
    )
  }

}

export default EditorHeader
