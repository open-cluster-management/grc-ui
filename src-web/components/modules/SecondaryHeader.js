/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import _ from 'lodash'
import { Breadcrumb, Tabs, Tab } from 'carbon-components-react'
import { Button, Tooltip } from '@patternfly/react-core'
import { createDisableTooltip } from '../common/DisableTooltip'
import resources from '../../../lib/shared/resources'
import { withRouter, Link } from 'react-router-dom'
import msgs from '../../../nls/platform.properties'
import checkCreatePermission from '../common/CheckCreatePermission'

resources(() => {
  require('../../../scss/secondary-header.scss')
})

export class SecondaryHeader extends React.Component {
  constructor(props) {
    super(props)
    this.renderBreadCrumb = this.renderBreadCrumb.bind(this)
    this.renderTabs = this.renderTabs.bind(this)
    this.renderLinks = this.renderLinks.bind(this)

    this.state = {
      shadowPresent: false
    }
  }

  componentDidMount() {
    window.addEventListener('scroll', this.listenToScroll)
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.listenToScroll)
  }

  listenToScroll = () => {
    if (window.scrollY > .1 && this.state.shadowPresent === false) {
      this.setState({shadowPresent:true})
    }
    else if (window.scrollY <= .1 && this.state.shadowPresent === true) {
      this.setState({shadowPresent:false})
    }
  }

  render() {
    const { tabs, title, breadcrumbItems, links, description, location, userAccess } = this.props
    const { shadowPresent: scrolled } = this.state
    const { locale } = this.context
    const displayType = location.pathname.split('/').pop()
    let showCreationLink // 0=clickable, 1=non-clickable, 2=hide
    switch(displayType) {
    case 'all':
    default:
      showCreationLink = checkCreatePermission(userAccess)
      break
    }
    const midName = (!location.pathname.startsWith('/multicloud/policies/all/') ? 'secondary-header-grc-overview' : '')
    const hasTabs = Boolean(tabs && tabs.length>0)
    const hasBreadcrumb = Boolean(breadcrumbItems)
    const hasButtons = Boolean(showCreationLink !== 2 && links && links.length>0)
    const noBreadcrumbClass = ' no-breadcrumb'
    return (
      <div
        className={`secondary-header-wrapper${scrolled?' scrolled':''}${hasTabs?'':' no-tabs'}${hasBreadcrumb?'':noBreadcrumbClass}${hasButtons?'':' no-buttons'}`}
        role='region'
        aria-label={title}
      >
        <div className={`secondary-header ${midName} simple-header${scrolled ? '-with-shadow' : ''}${description ? ' special-layout': ''}`}>
          <header aria-label={`Heading: ${title}`}>
            <div className='bx--detail-page-header-content'>
              {hasBreadcrumb &&
                (
                  <Breadcrumb>
                    {this.renderBreadCrumb()}
                  </Breadcrumb>
                )
              }
              {this.renderHeader(hasBreadcrumb, noBreadcrumbClass)}
              {hasTabs &&
                <Tabs selected={this.getSelectedTab() || 0} aria-label={`${title} ${msgs.get('tabs.label', locale)}`}>
                  {this.renderTabs()}
                </Tabs>
              }
            </div>
          </header>
        </div>
        {hasButtons &&
          <div className='secondary-header-links'>
            {this.renderLinks(showCreationLink)}
          </div>
        }
      </div>
    )
  }

  renderHeader(hasBreadcrumb = false, noBreadcrumbClass = ' no-breadcrumb') {
    const { title:headerTitle, description, information, links=[] } = this.props
    if (description) {
      /* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
      return (
        <div className={`bx--detail-page-header-title-container${hasBreadcrumb ? '': noBreadcrumbClass}`}>
          <h1 className='bx--detail-page-header-title'>{headerTitle}</h1>
          <div className='detail-page-header-title-button' onClick={description.action}><p>{description.display}</p></div>
        </div>
      )
    } else {
      return (
        <div className={`bx--detail-page-header-title-container${hasBreadcrumb ? '': noBreadcrumbClass}`}>
          <h1 className='bx--detail-page-header-title'>{headerTitle}</h1>
          {information &&
            <Tooltip content={information}>
              <svg className='info-icon'>
                <use href={'#diagramIcons_info'} ></use>
              </svg>
            </Tooltip>
          }
          {links && links.map(link => {
            const {id, kind, title:linkTitle } = link
            // if portal, react component will create the button using a portal
            if (kind==='portal' && linkTitle) {
              return <div key={id} id={id} className='portal' />
            } else {
              return null
            }})}
        </div>
      )
    }
  }

  renderBreadCrumb() {
    const { locale } = this.context
    const { breadcrumbItems } = this.props
    return breadcrumbItems && breadcrumbItems.filter(crumb => Boolean(crumb.label)).map((breadcrumb, index) => {
      const key = `${breadcrumb.id}-${index}`
      const noLocale = _.get(breadcrumb, 'noLocale', false)
      return (
        <div key={key} className='bx--breadcrumb-item' title={decodeURIComponent(breadcrumb.label)}>
          {breadcrumb.handleClick ?
            // eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events
            <div className='breadcrumb-link' onClick={breadcrumb.handleClick} >{msgs.get(breadcrumb.label, locale)}</div> :
            <Link to={breadcrumb.url || ''} className='bx--link'>{noLocale ? breadcrumb.label : msgs.get(breadcrumb.label, locale)}</Link>}
        </div>
      )
    })
  }

  renderTabs() {
    const { tabs } = this.props,
          { locale } = this.context
    return tabs.map(tab => {
      return <Tab label={msgs.get(tab.label, locale)} key={tab.id} id={tab.id} href={tab.url} onClick={tab.handleClick ? tab.handleClick : this.clickTab.bind(this, tab.url)} />
    })
  }

  renderLinks(showCreationLink) {
    const { links } = this.props,
          { locale } = this.context,
          disableFlag = (showCreationLink !== 1)
    return links.map(link => {
      const {id, label, url, kind='primary', handleClick=(()=> this.props.history.push(url)) } = link
      // if portal, react component will create the button using a portal
      if (kind==='portal') {
        return <div key={id} id={id} className='portal' />
      }
      const buttonItem = (<Button
        isDisabled={disableFlag}
        key={id}
        id={id}
        onClick={handleClick}
        variant={kind}
        isSmall
      >
        {msgs.get(label, locale)}
      </Button>)
      return createDisableTooltip(disableFlag, id, locale, buttonItem)
    })
  }

  getSelectedTab() {
    const { tabs, location } = this.props
    const selectedTab = tabs.map((tab, index) => {
      tab.index = index
      return tab
    }).filter((tab, index) => {
      if (index !== 0) {
        return location.pathname.startsWith(tab.url)
      }
      return undefined
    })
    return selectedTab[0] && selectedTab[0].index
  }

  clickTab(url) {
    this.props.history.push(url)
    return url
  }
}

SecondaryHeader.contextTypes = {
  locale: PropTypes.string
}

SecondaryHeader.propTypes = {
  breadcrumbItems: PropTypes.array,
  description: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  history: PropTypes.object.isRequired,
  information: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  links: PropTypes.array,
  location: PropTypes.object,
  tabs: PropTypes.array,
  title: PropTypes.string,
  userAccess: PropTypes.array
}

const mapStateToProps = (state) => {
  const userAccess = state.userAccess && state.userAccess.access
    ? state.userAccess.access
    : []
  return {
    title: state.secondaryHeader.title,
    tabs: state.secondaryHeader.tabs,
    breadcrumbItems: state.secondaryHeader.breadcrumbItems,
    links: state.secondaryHeader.links,
    refresh: state.secondaryHeader.refresh,
    userAccess: userAccess,
    description: state.secondaryHeader.description,
    information: state.secondaryHeader.information,
  }
}

export default withRouter(connect(mapStateToProps)(SecondaryHeader))
