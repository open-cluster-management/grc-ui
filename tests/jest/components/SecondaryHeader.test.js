/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
//curly braces means pure component without redux
//which is what we want in unit test
import { SecondaryHeader } from '../../../src-web/components/SecondaryHeader'
import { createMemoryHistory } from 'history'

const history = createMemoryHistory({'length':5,'action':'PUSH','location':{'pathname':'/multicloud/policies/all','search':'','hash':''}})

describe('SecondaryHeader component 1', () => {
  const location = {
    pathname: '/multicloud/policies'
  }
  it('renders as expected', () => {
    const component = renderer.create(
      <SecondaryHeader title='hello world' location={location} history={history} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('SecondaryHeader component 2', () => {
  const tabs = [{
    id: 'dashboard-application',
    label: 'tabs.dashboard.application',
    url: '/multicloud/dashboard',
  }]
  const location = {
    pathname: '/multicloud/policies/all'
  }
  it('renders clickable as expected', () => {
    const component = renderer.create(
      <SecondaryHeader
        title='hello world'
        tabs={tabs}
        location={location} history={history}
        links={[{'id':'create-policy','label':'button.create.policy','url':'/multicloud/policies/create'}]}
        userAccess={
          [
            {
              'namespace': 'calamari',
              'rules': {
                '*/*': [
                  '*'
                ]
              }
            },
          ]
        }
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders clickable as expected', () => {
    const component = renderer.create(
      <SecondaryHeader
        title='hello world'
        tabs={tabs}
        location={location} history={history}
        links={[{'id':'create-policy','label':'button.create.policy','url':'/multicloud/policies/create'}]}
        userAccess={
          [
            {
              'namespace': 'calamari',
              'rules': {
                'policy.open-cluster-management.io/policies': [
                  'get',
                  'list',
                  'watch',
                  'update',
                  'patch',
                  'create',
                  'delete',
                  'deletecollection'
                ]
              }
            },
          ]
        }
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders nonclickable as expected', () => {
    const component = renderer.create(
      <SecondaryHeader
        title='hello world'
        tabs={tabs}
        location={location} history={history}
        links={[{'id':'create-policy','label':'button.create.policy','url':'/multicloud/policies/create'}]}
        userAccess={
          [
            {
              'namespace': 'calamari',
              'rules': {
                'policy.open-cluster-management.io/policies': [
                  'get',
                  'list',
                  'watch',
                  'update',
                  'patch',
                ]
              }
            },
          ]
        }
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('SecondaryHeader component 3', () => {
  const tabs = [
    {
      id: 'logs-tab1',
      label: 'tabs.dashboard.application',
      url: '/multicloud/dashboard',
    },
    {
      id: 'logs-tab2',
      label: 'tabs.dashboard',
      url: '/hello',
    }
  ]
  const location = {
    pathname: '/multicloud/policies/findings'
  }
  it('renders as expected', () => {
    const component = renderer.create(

      // eslint-disable-next-line jsx-a11y/aria-role
      <SecondaryHeader title='hello world' role='Viewer' tabs={tabs} location={location} history={history} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('SecondaryHeader component 4', () => {
  const location = {
    pathname: '/multicloud/policies/findings'
  }
  it('renders as expected', () => {
    const component = renderer.create(
      <SecondaryHeader title='hello world' location={location} history={history} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('SecondaryHeader component 5', () => {
  const location = {
    pathname: '/multicloud/policies/findings'
  }
  it('clickTab as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history} />
    )
    expect(component.instance().clickTab('www.ibm.com')).toEqual('www.ibm.com')
  })
  it('renderHeader with description', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        description={{action:jest.fn(), display:jest.fn()}}
        location={location}
        history={history} />
    )
    expect(component.instance().renderHeader()).toMatchSnapshot()
  })
  it('renderHeader without description', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history} />
    )
    expect(component.instance().renderHeader()).toMatchSnapshot()
  })
  it('renderLinks nonclickable as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
        links={[{'id':'create-policy','label':'button.create.policy','url':'/multicloud/policies/create'}]}
      />
    )
    expect(component.instance().renderLinks()).toMatchSnapshot()
  })
  it('renderTabs as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
        tabs={[{'id':'grc-overview','label':'tabs.grc.overview','url':'/multicloud/policies','index':0},{'id':'grc-all','label':'tabs.grc.all','url':'/multicloud/policies/all','index':1},{'id':'grc-findings','label':'tabs.grc.findings','url':'/multicloud/policies/findings','index':2}]}
      />
    )
    expect(component.instance().renderTabs()).toMatchSnapshot()
  })

  it('renderBreadCrumb1 as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
      />
    )
    expect(component.instance().renderBreadCrumb()).toMatchSnapshot()
  })
  it('renderBreadCrumb2 as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
        breadcrumbItems={[]}
      />
    )
    expect(component.instance().renderBreadCrumb()).toMatchSnapshot()
  })
  it('renderBreadCrumb3 as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
        breadcrumbItems={[{id:'id1', label:'label1', url:'www.test1.com', handleClick:jest.fn(), noLocale:true}, {id:'id2', label:'label2', url:'www.test2.com', noLocale:true}, {id:'id3', label:'label3', url:'www.test3.com', handleClick:jest.fn(), noLocale:false}, {id:'id4', label:'label4', url:'www.test4.com', noLocale:false}]}
      />
    )
    expect(component.instance().renderBreadCrumb()).toMatchSnapshot()
  })
})
