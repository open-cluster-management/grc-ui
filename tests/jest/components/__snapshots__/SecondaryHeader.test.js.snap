// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryHeader component 1 renders as expected 1`] = `
<div
  aria-label="hello world secondary"
  className="secondary-header-wrapper-min"
  role="region"
>
  <div
    className="secondary-header simple-header"
  >
    <h1
      className="bx--detail-page-header-title"
    >
      hello world
    </h1>
  </div>
</div>
`;

exports[`SecondaryHeader component 2 renders clickable as expected 1`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
        <nav
          aria-label="hello world undefined"
          className="bx--tabs"
          role="navigation"
          selected={0}
        >
          <div
            aria-label="listbox"
            className="bx--tabs-trigger"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <a
              className="bx--tabs-trigger-text"
              href="#"
              onClick={[Function]}
              tabIndex={-1}
            >
              Application
            </a>
            <svg
              alt="show menu options"
              aria-label="show menu options"
              fillRule="evenodd"
              height="5"
              role="img"
              viewBox="0 0 10 5"
              width="10"
            >
              <title>
                show menu options
              </title>
              <path
                d="M0 0l5 4.998L10 0z"
              />
            </svg>
          </div>
          <ul
            className="bx--tabs__nav bx--tabs__nav--hidden"
            role="tablist"
          >
            <li
              className="bx--tabs__nav-item bx--tabs__nav-item--selected"
              id="dashboard-application"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={true}
              tabIndex={-1}
            >
              <a
                aria-selected={true}
                className="bx--tabs__nav-link"
                href="/multicloud/dashboard"
                role="tab"
                tabIndex={0}
              >
                Application
              </a>
            </li>
          </ul>
        </nav>
        <div
          aria-hidden={false}
          hidden={false}
          selected={true}
        />
      </div>
    </header>
  </div>
  <div
    className="secondary-header-links"
  >
    <button
      className="bx--btn bx--btn--primary"
      disabled={false}
      id="create-policy"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      Create policy
    </button>
  </div>
</div>
`;

exports[`SecondaryHeader component 2 renders clickable as expected 2`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
        <nav
          aria-label="hello world undefined"
          className="bx--tabs"
          role="navigation"
          selected={0}
        >
          <div
            aria-label="listbox"
            className="bx--tabs-trigger"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <a
              className="bx--tabs-trigger-text"
              href="#"
              onClick={[Function]}
              tabIndex={-1}
            >
              Application
            </a>
            <svg
              alt="show menu options"
              aria-label="show menu options"
              fillRule="evenodd"
              height="5"
              role="img"
              viewBox="0 0 10 5"
              width="10"
            >
              <title>
                show menu options
              </title>
              <path
                d="M0 0l5 4.998L10 0z"
              />
            </svg>
          </div>
          <ul
            className="bx--tabs__nav bx--tabs__nav--hidden"
            role="tablist"
          >
            <li
              className="bx--tabs__nav-item bx--tabs__nav-item--selected"
              id="dashboard-application"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={true}
              tabIndex={-1}
            >
              <a
                aria-selected={true}
                className="bx--tabs__nav-link"
                href="/multicloud/dashboard"
                role="tab"
                tabIndex={0}
              >
                Application
              </a>
            </li>
          </ul>
        </nav>
        <div
          aria-hidden={false}
          hidden={false}
          selected={true}
        />
      </div>
    </header>
  </div>
  <div
    className="secondary-header-links"
  >
    <button
      className="bx--btn bx--btn--primary"
      disabled={false}
      id="create-policy"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      Create policy
    </button>
  </div>
</div>
`;

exports[`SecondaryHeader component 2 renders nonclickable as expected 1`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
        <nav
          aria-label="hello world undefined"
          className="bx--tabs"
          role="navigation"
          selected={0}
        >
          <div
            aria-label="listbox"
            className="bx--tabs-trigger"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <a
              className="bx--tabs-trigger-text"
              href="#"
              onClick={[Function]}
              tabIndex={-1}
            >
              Application
            </a>
            <svg
              alt="show menu options"
              aria-label="show menu options"
              fillRule="evenodd"
              height="5"
              role="img"
              viewBox="0 0 10 5"
              width="10"
            >
              <title>
                show menu options
              </title>
              <path
                d="M0 0l5 4.998L10 0z"
              />
            </svg>
          </div>
          <ul
            className="bx--tabs__nav bx--tabs__nav--hidden"
            role="tablist"
          >
            <li
              className="bx--tabs__nav-item bx--tabs__nav-item--selected"
              id="dashboard-application"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={true}
              tabIndex={-1}
            >
              <a
                aria-selected={true}
                className="bx--tabs__nav-link"
                href="/multicloud/dashboard"
                role="tab"
                tabIndex={0}
              >
                Application
              </a>
            </li>
          </ul>
        </nav>
        <div
          aria-hidden={false}
          hidden={false}
          selected={true}
        />
      </div>
    </header>
  </div>
  <div
    className="secondary-header-links"
  >
    <button
      className="bx--btn bx--btn--primary bx--btn--disabled"
      disabled={true}
      id="create-policy"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      Create policy
    </button>
  </div>
</div>
`;

exports[`SecondaryHeader component 3 renders as expected 1`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
        <nav
          aria-label="hello world undefined"
          className="bx--tabs"
          role="navigation"
          selected={0}
        >
          <div
            aria-label="listbox"
            className="bx--tabs-trigger"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <a
              className="bx--tabs-trigger-text"
              href="#"
              onClick={[Function]}
              tabIndex={-1}
            >
              Application
            </a>
            <svg
              alt="show menu options"
              aria-label="show menu options"
              fillRule="evenodd"
              height="5"
              role="img"
              viewBox="0 0 10 5"
              width="10"
            >
              <title>
                show menu options
              </title>
              <path
                d="M0 0l5 4.998L10 0z"
              />
            </svg>
          </div>
          <ul
            className="bx--tabs__nav bx--tabs__nav--hidden"
            role="tablist"
          >
            <li
              className="bx--tabs__nav-item bx--tabs__nav-item--selected"
              id="logs-tab1"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={true}
              tabIndex={-1}
            >
              <a
                aria-selected={true}
                className="bx--tabs__nav-link"
                href="/multicloud/dashboard"
                role="tab"
                tabIndex={0}
              >
                Application
              </a>
            </li>
            <li
              className="bx--tabs__nav-item"
              id="logs-tab2"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={false}
              tabIndex={-1}
            >
              <a
                aria-selected={false}
                className="bx--tabs__nav-link"
                href="/hello"
                role="tab"
                tabIndex={-1}
              >
                provide a label
              </a>
            </li>
          </ul>
        </nav>
        <div
          aria-hidden={false}
          hidden={false}
          selected={true}
        />
        <div
          aria-hidden={true}
          hidden={true}
          selected={false}
        />
      </div>
    </header>
  </div>
</div>
`;

exports[`SecondaryHeader component 4 renders as expected 1`] = `
<div
  aria-label="hello world secondary"
  className="secondary-header-wrapper-min"
  role="region"
>
  <div
    className="secondary-header simple-header"
  >
    <h1
      className="bx--detail-page-header-title"
    >
      hello world
    </h1>
  </div>
</div>
`;

exports[`SecondaryHeader component 5 renderBreadCrumb1 as expected 1`] = `undefined`;

exports[`SecondaryHeader component 5 renderBreadCrumb2 as expected 1`] = `Array []`;

exports[`SecondaryHeader component 5 renderBreadCrumb3 as expected 1`] = `
Array [
  <div
    className="bx--breadcrumb-item"
    title="label1"
  >
    <div
      className="breadcrumb-link"
      onClick={[MockFunction]}
    />
  </div>,
  <div
    className="bx--breadcrumb-item"
    title="label2"
  >
    <Link
      className="bx--link"
      to="www.test2.com"
    >
      label2
    </Link>
  </div>,
  <div
    className="bx--breadcrumb-item"
    title="label3"
  >
    <div
      className="breadcrumb-link"
      onClick={[MockFunction]}
    />
  </div>,
  <div
    className="bx--breadcrumb-item"
    title="label4"
  >
    <Link
      className="bx--link"
      to="www.test4.com"
    />
  </div>,
]
`;

exports[`SecondaryHeader component 5 renderHeader with description 1`] = `
<div
  className="bx--detail-page-header-title-container"
>
  <h1
    className="bx--detail-page-header-title"
  >
    hello world
  </h1>
  <div
    className="detail-page-header-title-button"
    onClick={[MockFunction]}
  >
    <p>
      [MockFunction]
    </p>
  </div>
</div>
`;

exports[`SecondaryHeader component 5 renderHeader without description 1`] = `
<div
  className="bx--detail-page-header-title-container"
>
  <h1
    className="bx--detail-page-header-title"
  >
    hello world
  </h1>
</div>
`;

exports[`SecondaryHeader component 5 renderLinks nonclickable as expected 1`] = `
Array [
  <ForwardRef
    disabled={true}
    iconDescription="Provide icon description if icon is used"
    id="create-policy"
    kind="primary"
    onClick={[Function]}
    small={false}
    tabIndex={0}
    type="button"
  >
    Create policy
  </ForwardRef>,
]
`;

exports[`SecondaryHeader component 5 renderTabs as expected 1`] = `
Array [
  <Tab
    href="/multicloud/policies"
    id="grc-overview"
    label="Overview"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    selected={false}
    tabIndex={0}
  />,
  <Tab
    href="/multicloud/policies/all"
    id="grc-all"
    label="Policies"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    selected={false}
    tabIndex={0}
  />,
  <Tab
    href="/multicloud/policies/findings"
    id="grc-findings"
    label="Security findings"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    selected={false}
    tabIndex={0}
  />,
]
`;
